### ISSUE TITLE
C# Dev Kit unable to uninstall and C# files can not be opened

### ISSUE THREAD

**Issue by nanakonsta** [microsoft/vscode-dotnettools#712]
Type: <b>Bug</b>

C# Dev Kit unable to uninstall and C# files can not be opened

Extension version: 1.0.14
VS Code version: Code 1.84.2 (1a5daa3a0231a0fbba4f14db7ec463cf99d7768e, 2023-11-09T10:51:52.184Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|7.80GB (1.02GB free)|
|Process Argv|--crash-reporter-id 7009d260-90ce-4af5-9f8b-dc0c5d0da328|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263:30880072
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyonecf:30548226
282f8724:30602487
89544117:30613380
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30869155
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixtcf:30871695
pythonregdiag2:30871582
pyreplss1:30886140
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:30885898
pythontbext0:30879054
accentitlementst:30887150
dsvsc016cf:30886111
dsvsc017cf:30886113
dsvsc018:30886114
aa_t_chat:30882232

```

</details>

**Comment by timheuer** [comment]
Hi @nanakonsta are you still experiencing this? Any more information you can provide on error messages you are seeing?

**Comment by nanakonsta** [comment]
There was an older version file in programme files and I had to delete
everything and download it all over again. It took me hours to locate all
the related files by hand. Would be useful if the files would be updated
and replaced automatically every time we download a new version of Visual
Studio Code.

Στις Σάβ 16 Δεκ 2023 στις 1:46 π.μ., ο/η Tim Heuer ***@***.***>
έγραψε:

**Comment by timheuer** [comment]
Thanks @nanakonsta -- Sorry you had that. Each time an extension is updated the files are updated. If you are installing VSCode multiple times in different environments I recommend enabling Settings Sync where it will sync extensions/settings for you.

Okay to close this issue at this point?

**Comment by nanakonsta** [comment]
How do I enable that?


Στις Τετ 20 Δεκ 2023 στις 4:07 π.μ., ο/η Tim Heuer ***@***.***>
έγραψε:

**Comment by timheuer** [comment]
Check out https://code.visualstudio.com/docs/editor/settings-sync for instructions.

**Comment by nanakonsta** [comment]
This didn't help. Everything was already in sync and I am not a
professional developer to understand every little detail. The problem is
with dotnet. It is not recognised by Vs code although I have successfully
installed it.

Στις Τετ 20 Δεκ 2023 στις 4:47 μ.μ., ο/η Tim Heuer ***@***.***>
έγραψε:

**Comment by timheuer** [comment]
Ok sorry I had understood you got past that. 

Can you explain a little more? When you open a folder with a C# project are you seeing errors? If so which errors. In the output window is there anything in the C# and C# Dev Kit logs?  Can you also post the output of `dotnet —info` from a terminal window here?

happy to help narrow down the issue. Understanding what isn’t working specifically will help narrow it down.

**Comment by nanakonsta** [comment]
Hello and thank you for helping me out. It shows a message that it does not
recognise dotnet. I assure you I have downloaded dotnet and installed the
Dev kit in addition. I have also confirmed that through the CMD window
which I am sending you. I have installed dotnet 7 & 8 versions in the past
again and again because it always had issues and did not recognise  the C
sharp files only (no problem with python files for example). When I had
Visual studio it worked just fine and did not need any extra settings. Also
it was easy to connect with unity. Recently after all this trouble, I tried
uninstalling dotnet and Vs code and it did not let me because they were
connected with unity. It did not even let me to uninstall devkit
expenasion. So I suspect there is some kind of mistake with the path files
but I have no way to fix that. In addition I am scared to delete any files
with the name windows on them because they may be the system's files.

Yes I see errors. It has happened multiple times. In the terminal when I
write the dotnet console command it says that it can not do that because
dotnet does not exist. I found screenshots on the internet with the same
issue as me and the suggestions they were given was things I could not
understand. The problem being that I am a newbie. How can I find the logs
that you mention? Here are some pictures from my computer. I am currently
deleting dotnet c Sharp and unity to see whether there is a certain order
in which these three should be downloaded in order for them to function
properly.

Στις Πέμ 21 Δεκ 2023 στις 07:23 ο χρήστης Tim Heuer <
***@***.***> έγραψε:
